.circleci/osx_install_dependencies.sh

```diff
@@ -112,7 +112,7 @@ then
   rm -rf "$z3_dir"
 
   # evmone
-  evmone_version="0.11.0"
+  evmone_version="0.12.0"
```

Comment:
The checksum below needs to be updated as well. See line: https://github.com/ethereum/solidity/pull/15320/files#diff-95c60f439bb8a5efd25627b6a7c2b37310e42d8bd13e1cdb369fd3af41e88ea7R133

Reply:
I just checked the evmone repository and it seems that they don't ship `darwin-x86_64` anymore, since version `0.12.0`. In this case, as they are shipping `arm64` now (https://github.com/ethereum/evmone/releases/) we can remove the `if` below and update the checksum accordingly.

Reply:
Ah, great catch. @rodiazet can you please update?

Reply:
https://github.com/ethereum/solidity/pull/15320#discussion_r1756766505

Reply:
How did it even pass CI with the current checksum though? If we're not running CI on intel macs any more perhaps it's better to just remove it?

Reply:
We're ARM only (M1), so it was hitting the first branch and building from sources - the else branch is dead, and should be removed as @r0qs suggested. Or rather, the else branch should become the new standard and simply unpack from the tar.gz, since these are now available for ARM.

What I'm curious about is [this](https://app.circleci.com/pipelines/github/ethereum/solidity/35901/workflows/475d49a3-7b1f-44ed-8b1f-5f5df7a3a476/jobs/1635881). What does `b_bytecode_osx_intel-via-ir-optimize` even do, since we're running on ARM only? I assume this step should be renamed.

Reply:
Done

Reply:
> What I'm curious about is [this](https://app.circleci.com/pipelines/github/ethereum/solidity/35901/workflows/475d49a3-7b1f-44ed-8b1f-5f5df7a3a476/jobs/1635881). What does `b_bytecode_osx_intel-via-ir-optimize` even do, since we're running on ARM only? I assume this step should be renamed.

No, we have a universal binary, which contains both ARM and Intel code in one package and this step is testing that the Intel part works too. But IIRC it still runs on ARM and just runs under emulation.